rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_subscriber_info.hpp

```diff
@@ -147,6 +147,22 @@ class RMWSubscriptionEvent final : public EventListenerInterface
     const void * user_data,
     rmw_event_callback_t callback) override;
 
+  // void
+  // on_type_discovery(
+  //   DomainParticipant *,
+  //   const eprosima::fastrtps::rtps::SampleIdentity &,
+  //   const eprosima::fastrtps::string_255 & topic_name,
+  //   const eprosima::fastrtps::types::TypeIdentifier *,
+  //   const eprosima::fastrtps::types::TypeObject *,
+  //   eprosima::fastrtps::types::DynamicType_ptr dyn_type)) final
+  // {
+  // NOTE(methylDragon): The dynamic type deferred case is !! NOT SUPPORTED !!
+  //                     This is because currently subscriptions are required to have the type at
+  //                     construction to create the listener. Deferring it means that the listener
+  //                     construction will have to be deferred, and that would require logic changes
+  //                     elsewhere (e.g. to check for listener initialization status), which is
+  // }
+
```

Comment:
Should this just be removed then? I think (as I've stated elsewhere) we should just not use `NOTE`. I think this should either be a TODO or not left in.